documentation/src/main/asciidoc/topics/ref_role_permission_mappers.adoc

```diff
@@ -1,18 +1,54 @@
 [id='role-mappers_{context}']
 = Role and permission mappers
 
-{brandname} users are implemented with the `javax.security.auth.Subject` class and represent a grouping of security principals of type `java.security.Principal`.
+{brandname} users are implemented as a collection of principals. Principals represent either an individual user identity, such as a username,
+or a group to which the user belongs. Internally, these are implemented with the `javax.security.auth.Subject` class.
+
+In order to perform authorization, principals must be mapped to role names, which in turn are expanded into a set of permissions.
 
 {brandname} includes the `PrincipalRoleMapper` API to associate security principals to roles and the `RolePermissionMapper` API to associate roles with sets of permissions.
-{brandname} also provides the following role and permission mapper implementations:
+{brandname} provides the following role and permission mapper implementations:
```

Comment:
Isn't it better to say 'uses ** created by CLI/REST'

Reply:
These docs also applied to embedded, so no.